tests/test_dataset.py

```diff
@@ -168,6 +173,70 @@ def test_can_detect_and_import(self):
             self.assertEqual(imported_dataset.format, DEFAULT_FORMAT)
             compare_datasets(self, source_dataset, imported_dataset)
 
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_can_report_no_dataset_found(self):
+        env = Environment()
+        env.importers.items = {
+            DEFAULT_FORMAT: env.importers[DEFAULT_FORMAT],
+        }
+        env.extractors.items = {
+            DEFAULT_FORMAT: env.extractors[DEFAULT_FORMAT],
+        }
+
+        with TestDir() as test_dir, self.assertRaises(DatasetNotFoundError):
+            Dataset.import_from(test_dir, DEFAULT_FORMAT, env=env)
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_can_report_multiple_formats_match(self):
+        env = Environment()
+        env.importers.items = {
+            'a': env.importers[DEFAULT_FORMAT],
+            'b': env.importers[DEFAULT_FORMAT],
+        }
+        env.extractors.items = {
+            'a': env.extractors[DEFAULT_FORMAT],
+            'b': env.extractors[DEFAULT_FORMAT],
+        }
+
+        source_dataset = Dataset.from_iterable([
+            DatasetItem(id=1, annotations=[ Label(2) ]),
+        ], categories=['a', 'b', 'c'])
+
+        with TestDir() as test_dir, self.assertRaises(MultipleFormatsMatchError):
+            source_dataset.save(test_dir)
```

Comment:
IMO this should be rewritten so that the `save` call is outside of the `assertRaises` context. We need to check that it's specifically `import_from` that raises the exception.

Reply:
Changed